consensus/blake3pow/consensus.go

```diff
@@ -486,7 +486,15 @@ func (blake3pow *Blake3pow) CalcDifficulty(chain consensus.ChainHeaderReader, pa
 
 	if chain.IsGenesisHash(parent.Hash()) {
 		// Divide the parent difficulty by the number of slices running at the time of expansion
-		return parent.Difficulty()
+		if expansionNum == 0 && parent.Location().Equal(common.Location{}) {
+			// Base case: expansion number is 0 and the parent is the actual genesis block
+			return parent.Difficulty()
+		}
+		difficulty := new(big.Int).Exp(big.NewInt(2), parentEntropy, nil)
```

Comment:
This math is wrong because entropies are stored as bigBits.

bitsParentEntropy := new(big.Int).Sub(parentEntropy, big.NewInt(64))  //NOTE we need to use the precision constant from somewhere in the codebase

difficulty := new(big.Int).Exp(big.NewInt(2), bitsParentEntropy, nil)

Reply:
Please re-review, I used common.BigBitsToBits